tests/unit/mixins/route-test.js

```diff
@@ -284,61 +278,61 @@ test("It doesn't request more pages once _canLoadMore is false", function (asser
 
   this.createRoute();
 
-  assert.ok(this.route.get('_canLoadMore'), 'can load more');
-  assert.equal(this.route.get('currentPage'), 1, 'currentPage');
+  assert.ok(this.model.get('_canLoadMore'), 'can load more');
+  assert.equal(this.model.get('currentPage'), 1, 'currentPage');
 
   this.loadMore();
 
-  assert.notOk(this.route.get('_canLoadMore'), 'can load more');
-  assert.equal(this.route.get('currentPage'), 2, 'currentPage');
+  assert.notOk(this.model.get('_canLoadMore'), 'can load more');
+  assert.equal(this.model.get('currentPage'), 2, 'currentPage');
 
   this.loadMore();
 
-  assert.notOk(this.route.get('_canLoadMore'), 'can load more');
-  assert.equal(this.route.get('currentPage'), 2, 'currentPage');
+  assert.notOk(this.model.get('_canLoadMore'), 'can load more');
+  assert.equal(this.model.get('currentPage'), 2, 'currentPage');
 });
 
 test("It resets the currentPage when the model hook is called again", function (assert) {
   assert.expect(5);
 
   this.createRoute();
 
-  assert.ok(this.route.get('_canLoadMore'), 'can load more');
-  assert.equal(this.route.get('currentPage'), 1, 'currentPage');
+  assert.ok(this.model.get('_canLoadMore'), 'can load more');
+  assert.equal(this.model.get('currentPage'), 1, 'currentPage');
 
   this.callModelHook();
-  assert.equal(this.route.get('currentPage'), 1, 'currentPage');
+  assert.equal(this.model.get('currentPage'), 1, 'currentPage');
 
   this.loadMore();
 
-  assert.equal(this.route.get('currentPage'), 2, 'currentPage');
+  assert.equal(this.model.get('currentPage'), 2, 'currentPage');
 
   this.callModelHook();
 
-  assert.equal(this.route.get('currentPage'), 1, 'currentPage');
+  assert.equal(this.model.get('currentPage'), 1, 'currentPage');
 });
 
 module('RouteMixin - loading more data', {
   setup(assert) {
-    var store = createMockStore(Ember.Object.create({
-        items: [{id: 1, name: 'Test'}, {id: 2, name: 'Test 2'}],
-        pushObjects() {
-          return this;
-        },
-        meta: {
-          testTotalPages: 3
-        }
-      }),
+    let store = createMockStore(
+      // return response for model hook
+      EA([{id: 1, name: 'Test'}, {id: 2, name: 'Test 2'}], { meta: { testTotalPages: 3 } }),
+      // for query method in model hook
       (modelType, findQuery) => {
         assert.equal(findQuery.testPerPage, 1);
         assert.equal(findQuery.testPage, this.expectedPageNumber);
-    });
+      }
+    );
 
-    this.route = createRoute(['item', {perPage: 1, startingPage: 2}], {
+    this.route = createRoute(
+      ['item', {
+        // explicit params passed to infinityModel hook
```

Comment:
Must explicitly pass params via model hook because multiple models can exist on a route.  However, a fallback exists to lookup on route if needed (backwards compat or just the way it should be) - https://github.com/hhff/ember-infinity/pull/220/files#diff-95eb5c8003a4fd239be76137499e5176R129